core/websocket.go

```diff
@@ -0,0 +1,42 @@
+package core
+
+import (
+	"golang.org/x/net/websocket"
+)
+
+func queryWebSocket(endpoint *Endpoint, result *Result) {
+	origin := "http://localhost/"
+	url := endpoint.URL
+	message := endpoint.Body
+
+	config, err := websocket.NewConfig(url, origin)
+	if err != nil {
+		result.AddError("Error configuring WS connection:" + err.Error())
+		return
+	}
+
+	// Dial URL
+	ws, err := websocket.DialConfig(config)
+	if err != nil {
+		result.AddError("Error dialing WS:" + err.Error())
+		return
+	}
```

Comment:
When we create ws connection we should make sure to remove it once we finish using it
`defer ws.Close()`

Reply:
I thought Go's GC would take care of that?